flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/ProcessMetricGroup.java

```diff
@@ -18,18 +18,40 @@
 
 package org.apache.flink.runtime.metrics.groups;
 
+import org.apache.flink.api.common.time.Time;
 import org.apache.flink.runtime.metrics.MetricRegistry;
+import org.apache.flink.runtime.metrics.util.SystemResourcesCounter;
 
 /**
  * {@link AbstractImitatingJobManagerMetricGroup} implementation for process related metrics.
  */
 public class ProcessMetricGroup extends AbstractImitatingJobManagerMetricGroup {
 
+	private volatile SystemResourcesCounter systemResourcesCounter;
```

Comment:
I think this is not the right place. I would rather keep this part out of the `MetricGroup`.